scalafmt-core/shared/src/main/scala/org/scalafmt/config/SortSettings.scala

```diff
@@ -1,24 +1,34 @@
 package org.scalafmt.config
 
 import metaconfig._
+import scala.meta.classifiers.Classifier
+import scala.meta.Tree
+import scala.meta.Mod
+import sourcecode.Name
+import sourcecode.Text
 
 case class SortSettings(
     order: List[SortSettings.ModKey]
 )
 
 object SortSettings {
-
-  implicit val SortSettingsModKeyReader: ConfCodec[ModKey] =
-    ReaderUtil.oneOf[ModKey](
-      `implicit`,
-      `final`,
-      `sealed`,
-      `abstract`,
-      `override`,
-      `private`,
-      `protected`,
-      `lazy`
-    )
+  final case class ModKey(
+      name: String,
+      matches: Mod => Boolean
+  ) extends Product
+  private def createMod[T <: Mod](implicit
+      name: Name,
```

Comment:
also, question (on scala, probably): how does this work? why is there a unique implicit `Name` and `Classifier`?

Reply:
there is [macro machinery](https://github.com/lihaoyi/sourcecode/blob/master/sourcecode/src-2/sourcecode/Macros.scala#L72-L77) around it